documentation/documentation_publisher.ts

```diff
@@ -0,0 +1,110 @@
+import AWS from 'aws-sdk';
+import S3 from 'aws-sdk/clients/s3';
+import {promisify} from 'util';
+import {print, printAndExit} from '../testing/helpers';
+import {version} from '../package.json';
+import yargs, {Arguments} from 'yargs';
+import {exec as childExec} from 'child_process';
+const exec = promisify(childExec);
+
+const AwsRegion = 'us-west-2';
+const BaseGeneratedDocsPath = 'site';
+const DocumentationBucket = 'developer-documentation';
+const PacksSdkBucketRootPath = 'packs';
+const EnvironmentKeyRoot = `${PacksSdkBucketRootPath}/current`;
+const Environments = ['adhoc', 'head', 'staging', 'prod'];
+
+function handleError(e: Error) {
+  printAndExit(e.message + ` ${e.stack || ''}`);
+}
+
+function getS3Service(env: string): S3 {
+  const config: S3.ClientConfiguration = {
+    signatureVersion: 'v4',
+    region: AwsRegion,
+    credentials: new AWS.SharedIniFileCredentials({profile: env}),
+    computeChecksums: true,
+  };
+  return new S3(config);
+}
+
+function getS3Bucket(env: string): string {
+  return `coda-us-west-2-${env}-${DocumentationBucket}`;
+}
+
+function getS3ConfigKey(): string {
+  return `${PacksSdkBucketRootPath}/${version}`;
+}
+
+function getS3EnvironmentKey(environment: string): string {
+  return `${EnvironmentKeyRoot}/${environment}`;
+}
+
+async function pushDocsToEnv(env: string) {
+  const s3 = getS3Service(env);
+  const bucket = getS3Bucket(env);
+  const key = getS3ConfigKey();
+
+  async function pushDocsDirectory(): Promise<any> {
+    await exec(`aws s3 sync --profile ${env} --region ${AwsRegion} ${BaseGeneratedDocsPath} s3://${bucket}/${key}`);
```

Comment:
`spawnSync` may work here too, not sure if there's a need to make async vs not

Reply:
oh interesting didnt know a sync version existed will try that